win32ss/user/ntuser/kbdlayout.c

```diff
@@ -667,23 +667,27 @@ NtUserGetKeyboardLayoutName(
     _SEH2_TRY
     {
         ProbeForWriteUnicodeString(pustrName);
-        ProbeForWrite(pustrName->Buffer, pustrName->MaximumLength, 1);
+        ustrNameSafe = *pustrName;
+
+        ProbeForWrite(ustrNameSafe.Buffer, ustrNameSafe.MaximumLength, 1);
 
         if (IS_IME_HKL(pKl->hkl))
         {
-            RtlIntegerToUnicodeString((ULONG)(ULONG_PTR)pKl->hkl, 16, pustrName);
+            RtlIntegerToUnicodeString((ULONG)(ULONG_PTR)pKl->hkl, 16, &ustrNameSafe);
         }
         else
         {
-            if (pustrName->MaximumLength < KL_NAMELENGTH * sizeof(WCHAR))
+            if (ustrNameSafe.MaximumLength < KL_NAMELENGTH * sizeof(WCHAR))
             {
                 EngSetLastError(ERROR_INVALID_PARAMETER);
                 goto cleanup;
             }
             RtlInitUnicodeString(&ustrTemp, pKl->spkf->awchKF); /* FIXME: Do not use awchKF */
-            RtlCopyUnicodeString(pustrName, &ustrTemp);
+            RtlCopyUnicodeString(&ustrNameSafe, &ustrTemp);
```

Comment:
Instead of the RtlInitUnicodeString + RtlCopyUnicodeString calls, do instead:
```suggestion
            ustrNameSafe.Length = 0;
            RtlAppendUnicodeToString(&ustrNameSafe, pKl->spkf->awchKF); /* FIXME: Do not use awchKF */
```
Also, RtlAppendUnicodeToString returns an NTSTATUS, and you could do that to do error management.

Reply:
OK.